pkg/nar/reader_test.go

```diff
@@ -234,7 +234,10 @@ func TestReader(t *testing.T) {
 	}
 
 	hdr, err := p.Next()
+	// expect to return io.EOF at the end, and no more headers
 	assert.Nil(t, hdr)
-	// expect to be finished
 	assert.Equal(t, io.EOF, err)
+
+	assert.NoError(t, p.Close(), nil, "closing the reader shouldn't error")
+	assert.NoError(t, p.Close(), nil, "closing the reader multiple times shouldn't error")
```

Comment:
When an os.File is closed multiple times, it will err on the second call:

https://go.dev/play/p/LHDCVbhuF8J

Reply:
I changed the test to ensure it doesn't panic. It seems there's not really a convention on how closing multiple times should behave.